xamarin/xamarin.go

```diff
@@ -56,61 +58,69 @@ func (xamarinCmd CommandModel) GenerateArchive() (string, string, error) {
 
 // RunBuildCommand ...
 func (xamarinCmd CommandModel) RunBuildCommand() (string, string, error) {
-	// STO: https://stackoverflow.com/a/19534376/5842489
-	// if your project has a . (dot) in its name, replace it with a _ (underscore) when specifying it with /t
-	projectName := strings.Replace(xamarinCmd.ProjectName, ".", "_", -1)
-
-	archivesBeforeBuild, err := listArchives()
+	builder, err := builder.New(xamarinCmd.SolutionFilePath, []constants.SDK{constants.SDKIOS}, buildtools.Msbuild)
 	if err != nil {
-		return "", "", fmt.Errorf("failed to list archives, error: %s", err)
+		return "", "", err
 	}
 
-	cmdArgs := []string{constants.MsbuildPath,
-		xamarinCmd.SolutionFilePath,
-		fmt.Sprintf("/p:Configuration=%s", xamarinCmd.Configuration),
-		fmt.Sprintf("/p:Platform=%s", xamarinCmd.Platform),
-		fmt.Sprintf("/p:ArchiveOnBuild=true"),
-		fmt.Sprintf("/t:%s", projectName),
-	}
+	var outWriter bytes.Buffer
+	builder.SetOutputs(&outWriter, &outWriter)
 
-	log.Infof("$ %s", command.PrintableCommandArgs(true, cmdArgs))
-	cmd, err := command.NewFromSlice(cmdArgs)
-	if err != nil {
-		return "", "", fmt.Errorf("Failed to create Xamarin command, error: %s", err)
+	callback := func(solutionName string, projectName string, sdk constants.SDK, testFramwork constants.TestFramework, commandStr string, alreadyPerformed bool) {
```

Comment:
Could be defined outside of this function, and use an underscore `_` for not used variables.